> /hold Looking at the CI failures

```
Error: unknown flag: --v
Usage:
  /opt/vcloud/bin/cloud-director-named-disk-csi-driver [flags]

Flags:
      --cloud-config string   CSI driver cloud config
      --endpoint string       CSI endpoint
  -h, --help                  help for /opt/vcloud/bin/cloud-director-named-disk-csi-driver
      --nodeid string         node id
      --upgrade-rde           CSI upgrade rde
  -v, --version               version for /opt/vcloud/bin/cloud-director-named-disk-csi-driver

panic: error in executing command: [unknown flag: --v]
```

Flag has been removed so things should be good now.
/unhold

--- a/addons/csi/vmware-cloud-director/csi-controller.yaml
+++ b/addons/csi/vmware-cloud-director/csi-controller.yaml
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Sourced from:
-# - https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.5.0/manifests/csi-controller.yaml
+# - https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.6.0/manifests/csi-controller.yaml
 #
 # Changes:
 # - image source includes registry templating
@@ -105,6 +105,43 @@ subjects:
 roleRef:
   kind: ClusterRole
   name: csi-provisioner-role
+  apiGroup: rbac.authorization.k8s.io
+
+---
+# external resizer
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: csi-resizer-role
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["persistentvolumes"]
+    verbs: ["get", "list", "watch", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["persistentvolumeclaims"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["persistentvolumeclaims/status"]
+    verbs: ["update", "patch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: csi-resizer-binding
+subjects:
+  - kind: User
+    name: kubermatic:vcloud-csi
+    namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: csi-resizer-role
   apiGroup: rbac.authorization.k8s.io
 
 {{ end }}

--- a/addons/csi/vmware-cloud-director/csi-driver.yaml
+++ b/addons/csi/vmware-cloud-director/csi-driver.yaml
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Sourced from:
-# - https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.5.0/manifests/csi-driver.yaml
+# - https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.6.0/manifests/csi-driver.yaml
 
 {{ if eq .Cluster.CloudProviderName "vmwareclouddirector" }}
 

--- a/addons/csi/vmware-cloud-director/csi-node.yaml
+++ b/addons/csi/vmware-cloud-director/csi-node.yaml
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Sourced from:
-# - https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.5.0/manifests/csi-node.yaml
+# - https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.6.0/manifests/csi-node.yaml
 #
 # Changes:
 # - image source includes registry templating
@@ -78,7 +78,6 @@ spec:
           image: {{ Image "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.2.0" }}
           imagePullPolicy: "IfNotPresent"
           args:
-            - "--v=5"
             - "--csi-address=$(ADDRESS)"
             - "--kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)"
           lifecycle:
@@ -99,20 +98,32 @@ spec:
               mountPath: /csi
             - name: registration-dir
               mountPath: /registration
+        - name: csi-resizer
+          image: {{ Image "registry.k8s.io/sig-storage/csi-resizer:v1.4.0" }}
+          args:
+            - "--csi-address=$(ADDRESS)"
+            - "--timeout=30s"
+            - "--v=5"
+          env:
+            - name: ADDRESS
+              value: /var/lib/csi/sockets/pluginproxy/csi.sock
+          imagePullPolicy: "IfNotPresent"
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy/
         - name: vcd-csi-plugin
           securityContext:
             privileged: true
             capabilities:
               add: ["SYS_ADMIN"]
             allowPrivilegeEscalation: true
-          image: {{ Image "projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.5.0" }}
+          image: {{ Image "projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.6.0" }}
           imagePullPolicy: IfNotPresent
           command :
             - /opt/vcloud/bin/cloud-director-named-disk-csi-driver
             - --nodeid=$(NODE_ID)
             - --endpoint=$(CSI_ENDPOINT)
             - --cloud-config=/etc/kubernetes/vcloud/vcloud-csi-config.yaml
-            - --v=5
           env:
             - name: NODE_ID
               valueFrom:

--- a/addons/csi/vmware-cloud-director/vcloud-basic-auth.yaml
+++ b/addons/csi/vmware-cloud-director/vcloud-basic-auth.yaml
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Sourced from:
-# - https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.5.0/manifests/vcloud-basic-auth.yaml
+# - https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.6.0/manifests/vcloud-basic-auth.yaml
 
 {{ if eq .Cluster.CloudProviderName "vmwareclouddirector" }}
 

--- a/addons/default-storage-class/storage-class.yaml
+++ b/addons/default-storage-class/storage-class.yaml
@@ -211,6 +211,7 @@ reclaimPolicy: Delete
 parameters:
   storageProfile: {{ .Cluster.CSI.StorageProfile | quote }}
   filesystem: {{ default "ext4" .Cluster.CSI.Filesystem | quote }}
+allowVolumeExpansion: true
 {{ end }}
 {{ end }}
 

--- a/pkg/resources/cloudconfig/vmwareclouddirector/config.go
+++ b/pkg/resources/cloudconfig/vmwareclouddirector/config.go
@@ -27,7 +27,7 @@ import (
 	"k8c.io/kubermatic/v2/pkg/resources"
 )
 
-// Sourced from: https://raw.githubusercontent.com/vmware/cloud-director-named-disk-csi-driver/1.2.0/manifests/vcloud-csi-config.yaml
+// Sourced from: https://raw.githubusercontent.com/vmware/cloud-director-named-disk-csi-driver/1.6.0/manifests/vcloud-csi-config.yaml
 
 type CloudConfig struct {
 	VCD       VCDConfig `yaml:"vcd"`

--- a/pkg/resources/csi/vmwareclouddirector/deployment.go
+++ b/pkg/resources/csi/vmwareclouddirector/deployment.go
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Original source: https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.5.0/manifests/csi-controller.yaml
+// Original source: https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/1.6.0/manifests/csi-controller.yaml
 
 package vmwareclouddirector
 
@@ -33,7 +33,7 @@ import (
 )
 
 const (
-	csiVersion = "1.5.0"
+	csiVersion = "1.6.0"
 )
 
 var (
@@ -209,6 +209,44 @@ func ControllerDeploymentReconciler(data *resources.TemplateData) reconciling.Na
 					},
 				},
 				{
+					Name:            "csi-resizer",
+					ImagePullPolicy: corev1.PullIfNotPresent,
+					Image:           registry.Must(data.RewriteImage("registry.k8s.io/sig-storage/csi-resizer:v1.4.0")),
+					Args: []string{
+						"--csi-address=$(ADDRESS)",
+						"--timeout=30s",
+						"--kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig",
+						"--v=5",
+					},
+					Env: []corev1.EnvVar{
+						{
+							Name:  "ADDRESS",
+							Value: "/var/lib/csi/sockets/pluginproxy/csi.sock",
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "socket-dir",
+							MountPath: "/var/lib/csi/sockets/pluginproxy/",
+						},
+						{
+							Name:      resources.VMwareCloudDirectorCSIKubeconfigSecretName,
+							MountPath: "/etc/kubernetes/kubeconfig",
+							ReadOnly:  true,
+						},
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("10m"),
+							corev1.ResourceMemory: resource.MustParse("24Mi"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("100m"),
+							corev1.ResourceMemory: resource.MustParse("64Mi"),
+						},
+					},
+				},
+				{
 					Name:            "vcd-csi-plugin",
 					ImagePullPolicy: corev1.PullIfNotPresent,
 					Image:           registry.Must(data.RewriteImage("projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:" + csiVersion)),
@@ -216,7 +254,6 @@ func ControllerDeploymentReconciler(data *resources.TemplateData) reconciling.Na
 					Args: []string{
 						"--endpoint=$(CSI_ENDPOINT)",
 						"--cloud-config=/etc/kubernetes/vcloud/config",
-						"--v=5",
 					},
 					Env: []corev1.EnvVar{
 						{

--- a/pkg/resources/test/fixtures/deployment-vcd-1.27.0-csi-controller.yaml
+++ b/pkg/resources/test/fixtures/deployment-vcd-1.27.0-csi-controller.yaml
@@ -76,9 +76,32 @@ spec:
           name: vcloud-csi-kubeconfig
           readOnly: true
       - args:
+        - --csi-address=$(ADDRESS)
+        - --timeout=30s
+        - --kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig
+        - --v=5
+        env:
+        - name: ADDRESS
+          value: /var/lib/csi/sockets/pluginproxy/csi.sock
+        image: registry.k8s.io/sig-storage/csi-resizer:v1.4.0
+        imagePullPolicy: IfNotPresent
+        name: csi-resizer
+        resources:
+          limits:
+            cpu: 100m
+            memory: 64Mi
+          requests:
+            cpu: 10m
+            memory: 24Mi
+        volumeMounts:
+        - mountPath: /var/lib/csi/sockets/pluginproxy/
+          name: socket-dir
+        - mountPath: /etc/kubernetes/kubeconfig
+          name: vcloud-csi-kubeconfig
+          readOnly: true
+      - args:
         - --endpoint=$(CSI_ENDPOINT)
         - --cloud-config=/etc/kubernetes/vcloud/config
-        - --v=5
         command:
         - /opt/vcloud/bin/cloud-director-named-disk-csi-driver
         env:
@@ -88,7 +111,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
-        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.5.0
+        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.6.0
         imagePullPolicy: IfNotPresent
         name: vcd-csi-plugin
         resources:

--- a/pkg/resources/test/fixtures/deployment-vcd-1.28.0-csi-controller.yaml
+++ b/pkg/resources/test/fixtures/deployment-vcd-1.28.0-csi-controller.yaml
@@ -76,9 +76,32 @@ spec:
           name: vcloud-csi-kubeconfig
           readOnly: true
       - args:
+        - --csi-address=$(ADDRESS)
+        - --timeout=30s
+        - --kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig
+        - --v=5
+        env:
+        - name: ADDRESS
+          value: /var/lib/csi/sockets/pluginproxy/csi.sock
+        image: registry.k8s.io/sig-storage/csi-resizer:v1.4.0
+        imagePullPolicy: IfNotPresent
+        name: csi-resizer
+        resources:
+          limits:
+            cpu: 100m
+            memory: 64Mi
+          requests:
+            cpu: 10m
+            memory: 24Mi
+        volumeMounts:
+        - mountPath: /var/lib/csi/sockets/pluginproxy/
+          name: socket-dir
+        - mountPath: /etc/kubernetes/kubeconfig
+          name: vcloud-csi-kubeconfig
+          readOnly: true
+      - args:
         - --endpoint=$(CSI_ENDPOINT)
         - --cloud-config=/etc/kubernetes/vcloud/config
-        - --v=5
         command:
         - /opt/vcloud/bin/cloud-director-named-disk-csi-driver
         env:
@@ -88,7 +111,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
-        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.5.0
+        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.6.0
         imagePullPolicy: IfNotPresent
         name: vcd-csi-plugin
         resources:

--- a/pkg/resources/test/fixtures/deployment-vcd-1.29.0-csi-controller.yaml
+++ b/pkg/resources/test/fixtures/deployment-vcd-1.29.0-csi-controller.yaml
@@ -76,9 +76,32 @@ spec:
           name: vcloud-csi-kubeconfig
           readOnly: true
       - args:
+        - --csi-address=$(ADDRESS)
+        - --timeout=30s
+        - --kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig
+        - --v=5
+        env:
+        - name: ADDRESS
+          value: /var/lib/csi/sockets/pluginproxy/csi.sock
+        image: registry.k8s.io/sig-storage/csi-resizer:v1.4.0
+        imagePullPolicy: IfNotPresent
+        name: csi-resizer
+        resources:
+          limits:
+            cpu: 100m
+            memory: 64Mi
+          requests:
+            cpu: 10m
+            memory: 24Mi
+        volumeMounts:
+        - mountPath: /var/lib/csi/sockets/pluginproxy/
+          name: socket-dir
+        - mountPath: /etc/kubernetes/kubeconfig
+          name: vcloud-csi-kubeconfig
+          readOnly: true
+      - args:
         - --endpoint=$(CSI_ENDPOINT)
         - --cloud-config=/etc/kubernetes/vcloud/config
-        - --v=5
         command:
         - /opt/vcloud/bin/cloud-director-named-disk-csi-driver
         env:
@@ -88,7 +111,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
-        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.5.0
+        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.6.0
         imagePullPolicy: IfNotPresent
         name: vcd-csi-plugin
         resources:

--- a/pkg/resources/test/fixtures/deployment-vcd-1.30.0-csi-controller.yaml
+++ b/pkg/resources/test/fixtures/deployment-vcd-1.30.0-csi-controller.yaml
@@ -76,9 +76,32 @@ spec:
           name: vcloud-csi-kubeconfig
           readOnly: true
       - args:
+        - --csi-address=$(ADDRESS)
+        - --timeout=30s
+        - --kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig
+        - --v=5
+        env:
+        - name: ADDRESS
+          value: /var/lib/csi/sockets/pluginproxy/csi.sock
+        image: registry.k8s.io/sig-storage/csi-resizer:v1.4.0
+        imagePullPolicy: IfNotPresent
+        name: csi-resizer
+        resources:
+          limits:
+            cpu: 100m
+            memory: 64Mi
+          requests:
+            cpu: 10m
+            memory: 24Mi
+        volumeMounts:
+        - mountPath: /var/lib/csi/sockets/pluginproxy/
+          name: socket-dir
+        - mountPath: /etc/kubernetes/kubeconfig
+          name: vcloud-csi-kubeconfig
+          readOnly: true
+      - args:
         - --endpoint=$(CSI_ENDPOINT)
         - --cloud-config=/etc/kubernetes/vcloud/config
-        - --v=5
         command:
         - /opt/vcloud/bin/cloud-director-named-disk-csi-driver
         env:
@@ -88,7 +111,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
-        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.5.0
+        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.6.0
         imagePullPolicy: IfNotPresent
         name: vcd-csi-plugin
         resources:

--- a/pkg/resources/test/fixtures/deployment-vcd-1.31.0-csi-controller.yaml
+++ b/pkg/resources/test/fixtures/deployment-vcd-1.31.0-csi-controller.yaml
@@ -76,9 +76,32 @@ spec:
           name: vcloud-csi-kubeconfig
           readOnly: true
       - args:
+        - --csi-address=$(ADDRESS)
+        - --timeout=30s
+        - --kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig
+        - --v=5
+        env:
+        - name: ADDRESS
+          value: /var/lib/csi/sockets/pluginproxy/csi.sock
+        image: registry.k8s.io/sig-storage/csi-resizer:v1.4.0
+        imagePullPolicy: IfNotPresent
+        name: csi-resizer
+        resources:
+          limits:
+            cpu: 100m
+            memory: 64Mi
+          requests:
+            cpu: 10m
+            memory: 24Mi
+        volumeMounts:
+        - mountPath: /var/lib/csi/sockets/pluginproxy/
+          name: socket-dir
+        - mountPath: /etc/kubernetes/kubeconfig
+          name: vcloud-csi-kubeconfig
+          readOnly: true
+      - args:
         - --endpoint=$(CSI_ENDPOINT)
         - --cloud-config=/etc/kubernetes/vcloud/config
-        - --v=5
         command:
         - /opt/vcloud/bin/cloud-director-named-disk-csi-driver
         env:
@@ -88,7 +111,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
-        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.5.0
+        image: projects.registry.vmware.com/vmware-cloud-director/cloud-director-named-disk-csi-driver:1.6.0
         imagePullPolicy: IfNotPresent
         name: vcd-csi-plugin
         resources:

